synapse/cores/common.py

```diff
@@ -457,12 +457,23 @@ def _addDataModels(self, modtups):
 
             for fnam, fnfo, props in modl.get('forms', ()):
 
-                self.addRuntNode('syn:form', fnam, **fnfo)
-                self.addRuntNode('syn:prop', fnam, **fnfo)
+                dmpnfo = self.getPropDef(fnam)[1]
+                dmfnfo = dmpnfo.copy()
+                [dmfnfo.pop(key) for key, valu in list(dmpnfo.items()) if valu is None]
+                self._normTufoProps('syn:form', dmfnfo)
+                self.addRuntNode('syn:form', fnam, **dmfnfo)
+                dmfnfo = dmpnfo.copy()
+                [dmfnfo.pop(key) for key, valu in list(dmfnfo.items()) if valu is None]
+                self._normTufoProps('syn:prop', dmfnfo)
```

Comment:
_normTufoProps feels like it shouldn't be the responsibility of the caller of addRuntNode() (especially since addRuntNode *does* normalize any model props anyway?).  Was there something not working that mandated the fancy footwork?